lib/Basic/Platform.cpp

```diff
@@ -39,27 +39,54 @@ bool swift::tripleIsAnySimulator(const llvm::Triple &triple) {
       tripleIsAppleTVSimulator(triple);
 }
 
-StringRef swift::getPlatformNameForTriple(const llvm::Triple &triple) {
+DarwinPlatformKind swift::getDarwinPlatformKind(const llvm::Triple &triple) {
   if (triple.isiOS()) {
     if (triple.isTvOS()) {
       if (tripleIsAppleTVSimulator(triple))
-        return "appletvsimulator";
-      return "appletvos";
+        return DarwinPlatformKind::TvOSSimulator;
+      return DarwinPlatformKind::TvOS;
     }
 
     if (tripleIsiOSSimulator(triple))
-      return "iphonesimulator";
-    return "iphoneos";
+      return DarwinPlatformKind::IPhoneOSSimulator;
+    return DarwinPlatformKind::IPhoneOS;
   }
 
   if (triple.isWatchOS()) {
     if (tripleIsWatchSimulator(triple))
-        return "watchsimulator";
-    return "watchos";
+      return DarwinPlatformKind::WatchOSSimulator;
+    return DarwinPlatformKind::WatchOS;
   }
 
   if (triple.isMacOSX())
-    return "macosx";
+    return DarwinPlatformKind::MacOS;
+
+  llvm_unreachable("Unsupported Darwin platform");
+}
+
+static StringRef getPlatformNameForDarwin(const DarwinPlatformKind platform) {
+  switch(platform) {
+    case DarwinPlatformKind::MacOS:
```

Comment:
Nitpick: please don't indent switch cases, and please *do* put a space after the `switch` keyword.